.chronus/changes/examples-2024-5-14-18-46-26.md

```diff
@@ -0,0 +1,43 @@
+---
+# Change versionKind to one of: internal, fix, dependencies, feature, deprecation, breaking
+changeKind: feature
+packages:
+  - "@typespec/compiler"
+---
+
+Add new `@example` and `@opExample` decorator to provide examples on types and operations.
+
+  ```tsp
+  @example(#{
+    id: "some",
+    date: utcDateTime.fromISO("2020-01-01T00:00:00Z"),
+    timeout: duration.fromISO("PT1M"),
+  })
+  model Foo {
+    id: string;
+    date: utcDateTime;
+  
+    @encode("seconds", int32) timeout: duration;
+  }
+  ```
+  
+  ```tsp
+  @opExample(
+    #{
+      parameters: #{
+        pet: #{
+          id: "some",
+          name: "Fluffy",
+          dob: plainDate.fromISO("2020-01-01"),
+        },
+      },
+      returnType: #{
+        id: "some",
+        name: "Fluffy",
+        dob: plainDate.fromISO("2020-01-01"),
+      },
+    },
+    #{ title: "First", description: "Show creating a pet" }
```

Comment:
wondering here if we want to combine the parameters as one so you call this way instead

```
 @opExample(#{
      parameters: #{
        pet: #{
          id: "some",
          name: "Fluffy",
          dob: plainDate.fromISO("2020-01-01"),
        },
      },
      returnType: #{
        id: "some",
        name: "Fluffy",
        dob: plainDate.fromISO("2020-01-01"),
      },

   title: "First", 
   description: "Show creating a pet",
 })
```

Reply:
Although it is wordier, I think putting example metadata in a separate model example is a bit clearer